environment/os/install_deps.sh

```diff
@@ -0,0 +1,140 @@
+#!/bin/bash
+
+# Initialize a variable to track if the --build-deps or --toolchain-deps flag is passed
+build_deps_flag=false
+toolchain_deps_flag=false
+
+# Display help if no arguments are passed
+if [ "$#" -eq 0 ]; then
+    cat <<EOF
+Please specify one or both of these config flags:
+    --build-deps        for installing Memgraph build dependencies
+    --toolchain-deps    for installing toolchain runtime dependencies
+
+If you run into issues with this script, please run corresponding commands manually and open a github issue:
+    sudo ./environment/os/debian-10.sh install TOOLCHAIN_RUN_DEPS
+    sudo ./environment/os/debian-10.sh install MEMGRAPH_BUILD_DEPS
+    
+Link to create new github issue: https://github.com/memgraph/memgraph/issues/new?title=install-deps.sh%20...&assignee=gitbuda&body=%0A%0A%0A---%0AI%27m+a+human.+Please+be+nice.
+EOF
+    exit 0
+fi
+
+# Parse command-line arguments
+while [[ "$#" -gt 0 ]]; do
+    case $1 in
+        --build-deps) build_deps_flag=true ;;
+        --toolchain-deps) toolchain_deps_flag=true ;;
+        *) echo "Unknown parameter passed: $1"; exit 1 ;;
+    esac
+    shift
+done
+
+# Function to check and install dependencies for a given distribution script
+run_script() {
+    local distro_script=$1
+    echo "Running script for $distro_script"
+    if [ "$toolchain_deps_flag" = true ]; then
+        sudo ./environment/os/$distro_script check TOOLCHAIN_RUN_DEPS
+        sudo ./environment/os/$distro_script install TOOLCHAIN_RUN_DEPS
+    fi
+    if [ "$build_deps_flag" = true ]; then
+        sudo ./environment/os/$distro_script install MEMGRAPH_BUILD_DEPS
```

Comment:
Why is there no check for `MEMGRAPH_BUILD_DEPS` if you are checking for `TOOLCHAIN_RUN_DEPS`?

Reply:
I ended up refactoring script to:
- get rid of  `--build-deps` and `--toolchain-deps` args
- pass through all the args to` ./environment/os/$distro_scriptsh $@ `
- it's basically called like in these examples:
    `sudo ./install_deps.sh check TOOLCHAIN_RUN_DEPS`
    `sudo ./install_deps.sh install TOOLCHAIN_RUN_DEPS`
    `sudo ./install_deps.sh check MEMGRAPH_BUILD_DEPS` 
    `sudo ./install_deps.sh install MEMGRAPH_BUILD_DEPS`